lib/provider/data_types.go

```diff
@@ -98,26 +95,38 @@ type Volume struct {
 
 	// Only for VPC volume provider
 	VPCVolume
+
+	// ID of snapshot to be restored
+	SnapshotID string `json:"snapshotID,omitempty"`
 }
 
 // Snapshot ...
 type Snapshot struct {
-	Volume
+	VolumeID string `json:"volumeID"`
 
 	// a unique Snapshot ID which created by the provider
-	SnapshotID string `json:"snapshotID,omitempty"`
-
-	// The size of the snapshot, in GiB
-	SnapshotSize *int `json:"snapshotSize,omitempty"`
+	SnapshotID string `json:"snapshotID"`
 
-	// Source volume details
-	//SnapshotedVolume *Volume `json:"SnapshotedVolume"`
+	// The size of the snapshot, in bytes
+	SnapshotSize int64 `json:"snapshotSize"`
 
```

Comment:
Check int64

Reply:
```
size* | integerminimum: 1The size of this snapshot rounded up to the next gigabyte
```
So it is int64

Reply:
Volume Reference: https://github.com/IBM/ibmcloud-volume-vpc/blob/430f6c81c0a39b58b557b53e9569d9b786731d82/common/vpcclient/models/volume.go